test/test-fs.c

```diff
@@ -3203,28 +3203,54 @@ static void fs_read_bufs(int add_flags) {
   bufs[2] = uv_buf_init(scratch + 512, 128);
   bufs[3] = uv_buf_init(scratch + 640, 128);
 
-  ASSERT_EQ(446, uv_fs_read(NULL,
-                            &read_req,
-                            open_req1.result,
-                            bufs + 0,
-                            2,  /* 2x 256 bytes. */
-                            0,  /* Positional read. */
-                            NULL));
+
+  int res = uv_fs_read(NULL,
+                       &read_req,
+                       open_req1.result,
+                       bufs + 0,
+                       2,  /* 2x 256 bytes. */
+                       0,  /* Positional read. */
+                       NULL);
+#if defined(__CYGWIN__) ||                                                    \
+    defined(__MVS__) ||                                                       \
+    (defined(__HAIKU__) && B_HAIKU_VERSION < B_HAIKU_VERSION_1_PRE_BETA_5) || \
+    (defined(__sun) && !defined(__illumos__))
```

Comment:
I'm going to guess you blindly copied these conditionals without putting much thought into whether they apply to other platforms... sloppy, please fix that and pay closer attention going forward.

I care 0% about zos so if IBM wants non-cross-platform behavior, that's fine by me, but it'd be best to fix up uv_fs_read() rather than the test suite.